core/tools/cpu.py

```diff
@@ -69,35 +73,44 @@ def read(self, inputs: str) -> str:
         "ex. test.py\nprint('hello world')\n "
         "and the output will be last 3 line.",
     )
-    def write(self, inputs: str) -> str:
+    def append(self, inputs: str) -> str:
         try:
             code = CodeWriter.append(inputs)
-            output = "Last 3 line was:\n" + "\n".join(code.split("\n")[-3:])
+            output = (
+                "Last 3 line was:\n"
+                + "\n".join(code.split("\n")[-3:])
+                + "\nYou can use CodeEditor.APPEND tool to append the code if it is not completed."
+            )
         except Exception as e:
             output = str(e)
 
         logger.debug(
-            f"\nProcessed CodeEditor, Input: {inputs} " f"Output Answer: {output}"
+            f"\nProcessed CodeEditor.APPEND, Input: {inputs} "
+            f"Output Answer: {output}"
         )
         return output
 
     @tool(
         name="CodeEditor.WRITE",
         description="Write code to create a new tool. "
         "If the code is completed, use the Terminal tool to execute it, if not, append the code through the CodeEditor.APPEND tool. "
-        "Input should be filename and code. "
+        "Input should be filename and code. Filename should be start with './playground/'. "
```

Comment:
what if it trys to write `./playground/..` ? shouldn't it have hardly coded rules not only prompt?

such as: 
```python
if not str(Path("filepath").resolve()).startswith(Path(".").resolve()):
    raise Exception("nah")
```

Reply:
I agree. I didn't think of that.